source/_posts/marionette-v3.md

```diff
@@ -0,0 +1,284 @@
+title: Marionette.js v3. Getting started with new version 
+date: 2016-08-10 21:19:00 
+tags: 
+- backbone
+- marionette
+- learning
+- v3
+
+---
+
+Welcome to the party.
+
+The main goal of Marionette v3 was to clean up and simplify the API.
+The library was converted to ES6 and organized for easier contribution.
+Marionette v3 is both smaller and faster than v2.x.
+
+In future major releases we hope to improve the performance and simplify further.
+
+Let's go through some of the changes in v3.
+
+## View
+
+Version 2.x had many different kinds of views: `View`, `ItemView`,
+`LayoutView`, `CollectionView`, `CompositeView`.
+
+In version 3 `ItemView`, and `LayoutView` were 'merged' into `View`,
+and `CompositeView` was deprecated for removal in v4.
+Now we have only `View` and `CollectionView`.
+
+Here are some simple examples:
+
+````js
+import Mn from 'backbone.marionette';
+
+const template = _.template('<h1>Marionette says hello!</h1>');
+const MyView = Mn.View.extend({
+  el: 'body',
+  template: template
+});
+const myView = new MyView();
+
+myView.render();
+````
+
+[Live example](https://jsfiddle.net/marionettejs/1myrqxtq/)
+
+
+#### How does `View` handle regions?
+The same as `LayoutView` did however regions are no longer attached 
+directly to the view.
+
+````js
+import Mn from 'backbone.marionette';
+
+const template1 = _.template('<h1>Marionette says hello!</h1>');
+const template2 = _.template('<h1>Marionette is awesome!</h1>');
+
+const myView1 = new Mn.View({template: template1});
+const myView2 = new Mn.View({template: template2});
+
+const MyView = Mn.View.extend({
+  el: '#container',
+  template: false,
+  regions: {
+    region1: '#region1',
+    region2: '#region2'
+  },
+
+  onRender() {
+    // Since in v3 regions are no longer attached to the view
+    // this.region1.show(myView1) will throw an error saying 
+    // show is not a function on undefined (region1).
+    this.getRegion('region1').show(myView1);
+    this.showChildView('region2', myView2); // Preferred method for showing child views
+  }
+});
+const myView = new MyView();
+
+myView.render();
+````
+
+[Live example](https://jsfiddle.net/marionettejs/bux7fppo/)
+
+
+#### How can I build table or tree without `CompositeView`?
+
+If you build your table or tree using `CompositeView` just take a look [here](http://marionettejs.com/docs/v3.0.0/marionette.collectionview.html#rendering-tables)
+to learn how to easily migrate to using `View` and `CollectionView`.
+
+
+### View events
+
+In version 3 the region events `show` and `before:show` are no longer triggered
+on the view. You can use `render` and `before:render` events in most cases. If
+you need to know that the view is in the DOM then you can use `attach` or `dom:refresh`
+which is triggered after the attach event and every _subsequent_ `render` event.
+Learn more about view lifecycle events [here](http://marionettejs.com/docs/v3.0.0/marionette.view.html#lifecycle-events).
```

Comment:
@paulfalgout I think this link has to be changed because of https://github.com/marionettejs/backbone.marionette/pull/3120